pir/pir-internal/src/main/java/com/duckduckgo/pir/internal/optout/PirOptOut.kt

```diff
@@ -173,26 +217,27 @@ class RealPirOptOut @Inject constructor(
             cleanRunners()
             runners.clear()
         }
-        obtainProfile()
+        obtainProfiles()
 
-        logcat { "PIR-OPT-OUT: Running opt-out on profile: $profileQuery on ${Thread.currentThread().name}" }
+        logcat { "PIR-OPT-OUT: Running opt-out on profiles: $profileQueries on ${Thread.currentThread().name}" }
 
         val script = pirCssScriptLoader.getScript()
 
-        val brokerSteps = brokers.mapNotNull { broker ->
-            repository.getBrokerOptOutSteps(broker)?.run {
-                brokerStepsParser.parseStep(broker, this)
-            }
-        }.filter {
-            it.isNotEmpty()
-        }.flatten()
-
-        maxWebViewCount = if (brokerSteps.size <= MAX_DETACHED_WEBVIEW_COUNT) {
-            brokerSteps.size
-        } else {
-            MAX_DETACHED_WEBVIEW_COUNT
+        // Load opt-out steps jsons for each broker
+        val brokerOptOutStepsJsons = brokers.mapNotNull { broker ->
+            repository.getBrokerOptOutSteps(broker)?.let { broker to it }
         }
 
+        // Create a map of profiles that have opt-out steps for those brokers
+        val profileBrokerStepsMap = profileQueries.associateWith { profileQuery ->
+            brokerOptOutStepsJsons.map { (broker, stepsJson) ->
+                brokerStepsParser.parseStep(broker, stepsJson, profileQuery)
+            }.flatten()
+        }.filter { it.value.isNotEmpty() }
+
+        val stepsCount = profileBrokerStepsMap.values.maxOf { it.size }
```

Comment:
This logic seems incorrect. The thing that we want to do here is to use the total number of profile x broker x extractedProfile instances IF it’s lesser than the max limit. In this case, we are just taking the highest value.